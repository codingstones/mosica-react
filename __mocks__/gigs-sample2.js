export const sampleGigs = [
  {
    'day': 'Sábado, 20 de Agosto',
    'gigs': [
      {
        'title': 'rapsus',
        'lat_lng': '41.641851935961654,-0.8751129897638315',
        'address': 'cesareo alierta, 5001,',
        'description': 'Concierto de rapsus',
        'price': '',
        'image_url': 'http://www.zaragoza.es/cont/paginas/actividades/imagen/web_320x480px.png',
        'affiliate_url': null,
        'place': 'Avenida Cesáreo Alierta (mobilityweek)',
        'slug': 'javat',
        'schedule_id': 5280,
        'hour': '19:30',
        'day': '2017-09-18',
        'schedule': [
          {
            'id': 5280,
            'day': '2017-09-18',
            'hour': '19:30',
            'gig_id': 5927,
            'created_at': '2017-09-07T15:57:35.419Z',
            'updated_at': '2017-09-07T15:57:35.419Z'
          }
        ]
      }
    ]
  }
]
