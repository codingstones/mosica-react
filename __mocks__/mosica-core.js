import { sampleGigs } from './gigs-sample'
const mosicaCore = jest.genMockFromModule('mosica-core')

class GigService {
  retrieveNextGigs() {
    return Promise.resolve(sampleGigs)
  }
}

mosicaCore.GigService = GigService
module.exports = mosicaCore
