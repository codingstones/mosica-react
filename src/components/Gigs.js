import React from 'react';

export const Gigs = (props) => {
  const {gigs, onClick} = props;

  return (
    <div>
      {gigs.map((gig) =>
        <div className="row" key={gig.id}
             onClick={onClick(gig.id)}>
             {gig.title}-{gig.place}
        </div>
      )}
    </div>
  );
};