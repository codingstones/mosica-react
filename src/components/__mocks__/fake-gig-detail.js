export const fakeGigDetail = {
  address: "Avda. Cesáreo Alierta, 120, 50013, Zaragoza",
  affiliate_url: null,
  day: "2017-12-16",
  description:
    "Enrique Bunbury ha anunciado el lanzamiento de su nuevo disco de estudio Expectativas. Hará una parada en Zaragoza el próximo 16 de diciembre para presentarlo ante sus paisanos. Este nuevo álbum es la continuación de Palosanto de 2013. La espera ha sido de cuatro años, pero el 8 de septiembre llegarán por fin los dos primeros adelantos,",
  hour: "22:00",
  id: 6533,
  image_url: "http://www.zaragoza.es/cont/paginas/actividades/imagen/pfotoevento4020.png",
  lat_lng: "41.63602275205834,-0.8658168875758695",
  place: "Pabellón Príncipe Felipe",
  price: "",
  schedule: [],
  schedule_id: 9119,
  slug: "concierto-enrique-bunbury",
  title: "Concierto Enrique Bunbury"
}