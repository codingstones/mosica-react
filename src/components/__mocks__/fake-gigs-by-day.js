export const fakeGigsByDay = [
  {
    'day': 'Lunes, 18 de Septiembre',
    'gigs': [
      {
        'id': '123456',
        'title': 'Anarchy in the JS',
        'lat_lng': '41.641851935961654,-0.8751129897638315',
        'address': 'BCN',
        'description': 'Coding Stones GigRow',
        'price': '',
        'image_url': 'http://www.zaragoza.es/cont/paginas/actividades/imagen/web_320x480px.png',
        'affiliate_url': null,
        'place': 'SCBCN',
        'slug': 'stones',
        'schedule_id': 5280,
        'hour': '19:30',
        'day': '2017-09-18',
        'schedule': [
          {
            'id': 5280,
            'day': '2017-09-18',
            'hour': '19:30',
            'gig_id': 5927,
            'created_at': '2017-09-07T15:57:35.419Z',
            'updated_at': '2017-09-07T15:57:35.419Z'
          }
        ]
      },
      {
        'id': '2222222',
        'title': 'Anarchy in the Codemotion',
        'lat_lng': '41.641851935961654,-0.8751129897638315',
        'address': 'BCN',
        'description': 'Another Coding Stones GigRow',
        'price': '',
        'image_url': 'http://www.wtf.es/wtf.png',
        'affiliate_url': null,
        'place': 'Around the world',
        'slug': 'stones',
        'schedule_id': 5280,
        'hour': '19:30',
        'day': '2017-12-03',
        'schedule': [
          {
            'id': 5280,
            'day': '2017-09-18',
            'hour': '19:30',
            'gig_id': 5927,
            'created_at': '2017-09-07T15:57:35.419Z',
            'updated_at': '2017-09-07T15:57:35.419Z'
          }
        ]
      },
    ]
  }
]

export const FIRST_DAY_GIG_TEXTS = fakeGigsByDay[0].gigs.map(gig => gig.title)
export const FIRST_GIG = fakeGigsByDay[0].gigs[0]
