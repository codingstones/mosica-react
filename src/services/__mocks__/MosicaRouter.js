const navigateToGigMock = jest.fn()
const RouterMock = { navigateToGig: navigateToGigMock}

const MosicaRouter = (history)=> {
  return RouterMock;
}
module.exports = {
  MosicaRouter
}

