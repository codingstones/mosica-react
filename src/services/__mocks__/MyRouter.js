const navigateToGigMock = jest.fn()
const RouterMock = { navigateToGig: navigateToGigMock}

const MyRouter = (history)=> {
  return RouterMock;
}
module.exports = {
  MyRouter: MyRouter,
}

