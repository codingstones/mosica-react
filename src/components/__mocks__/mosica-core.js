import { fakeGigsByDay } from './fake-gigs-by-day'
const mosicaCore = jest.genMockFromModule('mosica-core')

class GigService {
  retrieveNextGigs() {
    return Promise.resolve(fakeGigsByDay)
  }
}

mosicaCore.GigService = GigService
module.exports = mosicaCore
