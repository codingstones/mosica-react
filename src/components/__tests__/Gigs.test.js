import React from 'react';
import renderer from 'react-test-renderer';
import {fakeGigsByDay} from '../__mocks__/fake-gigs-by-day';
import { Gigs } from '../Gigs';

it('renders GigsDay', () => {
  const component = renderer.create(
    <Gigs gigs={fakeGigsByDay[0].gigs} onClick={jest.fn}/>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

