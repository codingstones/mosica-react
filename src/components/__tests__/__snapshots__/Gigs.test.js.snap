// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders GigsDay 1`] = `
<div>
  <div
    className="row"
    onClick={[Function]}
  >
    Anarchy in the JS
    -
    SCBCN
  </div>
  <div
    className="row"
    onClick={[Function]}
  >
    Vegan Microservices
    -
    Around the world
  </div>
</div>
`;
